### PR TITLE
[BugFix] Should use the column of create table, not modify column of alter clause in the modify column process of schema change.

### LINKED ISSUE #9016: [Bug] Should use the column of create table, not modify column of alter clause in the modify column process of schema change.
### Steps to reproduce the behavior (Required)
```
create table orders (
    dt date NOT NULL,
    order_id bigint NOT NULL,
    user_id int NOT NULL,
    merchant_id int NOT NULL,
    good_id int NOT NULL,
    good_name string NOT NULL,
    price int NOT NULL,
    cnt int NOT NULL,
    revenue int NOT NULL,
    state tinyint NOT NULL
) PRIMARY KEY (dt, order_id)
DISTRIBUTED BY HASH(order_id) BUCKETS 4
PROPERTIES("replication_num" = "1");
```
```
mysql> alter table orders modify column dt largeint;
```

### Expected behavior (Required)

should report the error directly, clearly

```
mysql> alter table orders modify column dt largeint;
ERROR 1064 (HY000): Can not modify key column: dt for primary key table
```

### Real behavior (Required)

but the real world error report do not make sense

```
mysql> alter table orders modify column dt largeint;
ERROR 1064 (HY000): Invalid column order. value should be after key. index[orders]
```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

mysql> select current_version();
```
+-------------------+
| current_version() |
+-------------------+
| UNKNOWN cf0fb08a7 |
+-------------------+
1 row in set (0.00 sec)
```

main

### PR DESCRIPTION
**Pull request by Linkerist**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #9016

### CODE REVIEW

**Comment by Linkerist** [comment]
run starrocks_fe_unittest

File: fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java [hunk 1]
```diff
@@ -325,37 +325,38 @@ private void processDropColumn(DropColumnClause alterClause, OlapTable olapTable
...
-                        modColumn.getName());
+                        baseColumn.getName());
             }
             modColumn.setAggregationType(AggregateType.REPLACE, true);
         } else if (KeysType.AGG_KEYS == olapTable.getKeysType()) {
-            if (modColumn.isKey() && null != modColumn.getAggregationType()) {
-                throw new DdlException("Can not assign aggregation method on key column: " + modColumn.getName());
-            } else if (null == modColumn.getAggregationType()) {
-                Preconditions.checkArgument(modColumn.getType().isScalarType());
+            if (baseColumn.isKey() && null != baseColumn.getAggregationType()) {
```
Comment (gengjun-git) [inline fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java:339]: There should be modColumn.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 0 / 3 (00.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/alter/SchemaChangeHandler.java|0|3|00.00%|[329, 348, 356]|

File: fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java (hunk 1 above)
Comment (Linkerist) [reply fe/fe-core/src/main/java/com/starrocks/alter/SchemaChangeHandler.java]: yes, changed.

**Review by chaoyli — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**